Demo/Demo/WebServices/Alamofire/Views/AFProductViewController.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  ProductViewController.swift
+//  Demo
+//
+//  Created by Krunal Patel on 15/06/23.
+//
+
+import UIKit
+
+protocol AFProductViewControllerDelegate: AnyObject {
+    func product(filtersDidChange filters: [String])
+    func product(productsDidChange products: [Product])
+}
+
+class AFProductViewController: UIViewController {
+    
+    // MARK: - Outlets
+    @IBOutlet private weak var collectionViewFilter: UICollectionView!
+    @IBOutlet private weak var collectionViewProduct: UICollectionView!
+    
+    // MARK: - Variables
+    private var viewModel: ProductViewModel!
+    
+    override func viewDidLoad() {
+        viewModel = ProductViewModel()
```

Comment:
no need to initialize on did load you can directly initialize at the time of creating object of ProductViewModel()